master/buildbot/test/fake/fakedb.py

```diff
@@ -1655,7 +1656,7 @@ def setState(self, objectid, name, value):
 
     def atomicCreateState(self, objectid, name, thd_create_callback):
         value = thd_create_callback()
-        self.states[objectid][name] = json.dumps(value)
+        self.states[objectid][name] = json.dumps(bytes2NativeString(value))
```

Comment:
Well, that's your opinion.  To have a working branch, I made the change as I did.
Without the change I did, things break.
Hunting down all the callers and making sure they are doing things right with native string
is a lot of work, with potential to break more things in other places.